docs/ref/utils.txt

```diff
@@ -704,6 +704,19 @@ escaping HTML.
 
         format_html_join("\n", "<li>{} {}</li>", ((u.first_name, u.last_name) for u in users))
 
+    ``args_generator`` can also be an iterator that returns a sequence of mappings
+    that will be passed to :func:`format_html` as keyword arguments. For example::
+
+        format_html_join(
+            "\n",
+            "<li>{first_name} {last_name}</li>",
+            ({"first_name": u.first_name, "last_name": u.last_name} for u in users),
+        )
```

Comment:
Great start here. I think it’s a bit unclear to leave line 702 saying that it “should” be for positional arguments, but then later say “it can also be...”. Let’s try describe `args_generator` fully before giving examples. I suggest we edit from line 702 down to 714 to say:

```rst
``args_generator`` should be an iterator that yields arguments to pass to
:func:`format_html`, either sequences of positional arguments or mappings of
keyword arguments.

For example, tuples can be used for positional arguments::

    format_html_join(
        "\n",
        "<li>{} {}</li>",
        ((u.first_name, u.last_name) for u in users),
    )

Or dictionaries can be used for keyword arguments::

    format_html_join(
        "\n",
        '<li data-id="{id}">{id} {title}</li>',
        ({"id": b.id, "title": b.title} for b in books),
    )
```

I also adjusted the second example to show the motivating example of reusing an argument in multiple places within the template string.